libs/langgraph/tests/test_prebuilt.py

```diff
@@ -368,6 +374,39 @@ def tool2(some_val: int) -> str:
         create_react_agent(model.bind_tools([tool1]), [tool2])
 
 
+def test__infer_handled_types() -> None:
+    def handle(e):  # type: ignore
+        return ""
+
+    def handle2(e: Exception) -> str:
+        return ""
+
+    def handle3(e: Union[ValueError, ToolException]) -> str:
+        return ""
+
+    class Handler:
+        def handle(self, e: ValueError) -> str:
+            return ""
+
+    handle4 = Handler().handle
+
+    expected: tuple = (Exception,)
+    actual = _infer_handled_types(handle)
+    assert expected == actual
+
+    expected = (Exception,)
+    actual = _infer_handled_types(handle2)
+    assert expected == actual
+
+    expected = (ValueError, ToolException)
+    actual = _infer_handled_types(handle3)
+    assert expected == actual
+
+    expected = (ValueError,)
+    actual = _infer_handled_types(handle4)
+    assert expected == actual
+
+
 async def test_tool_node():
```

Comment:
This is a unit test with 300 lines of stuff in it? Any way to break apart? It's hard to follow right now to understand if the API works as expected